client/src/components/RescheduleModal/RescheduleModal.jsx

```diff
@@ -84,23 +121,46 @@ export default class RescheduleModal extends React.PureComponent {
         );
     };
 
-    _renderContent = () => {
+    _onDateSelection = (value) => {
         const {bookingsListStore} = this.props;
         const {rescheduleModal} = bookingsListStore.store;
-        const {loading, data, slots, errorMessage} = rescheduleModal;
+        this.setState({selectedDate: value});
+        bookingsListStore.setRescheduleModalData('slots', null);
+        bookingsListStore.setRescheduleModalData('selectedSlot', null);
+        bookingsListStore.fetchScheduleSlots(rescheduleModal.data.bookedEntity.scheduleId, moment(value).add(START_HOUR_OF_DAY, 'h').toISOString(), moment(value).add(24, 'h').toISOString());
+    };
 
+    _displayAfterDateSelection = () => {
+        const {bookingsListStore} = this.props;
+        const {rescheduleModal} = bookingsListStore.store;
+        const {loading, slots} = rescheduleModal;
+        const {selectedDate} = this.state;
+        if (selectedDate) {
+            return loading && !slots ? this._renderSlotsSkeleton() : this._renderSlots();
+        }
```

Comment:
please return a null value as default
OR
consult with @alonroth 
perhaps there should be a text that says you need to select a date first